src/main/scala/tech/christopherdavenport/twitterstorm/Resource.scala

```diff
@@ -0,0 +1,47 @@
+package tech.christopherdavenport.twitterstorm
+
+import cats.effect.{Async, Effect}
+import fs2.Stream
+
+abstract class Resource[F[_], A] { self =>
```

Comment:
This is like a `ContT` monad, but specialized for `Stream`, which we know has safe semantics for cleanup.